components/state.js

```diff
@@ -0,0 +1,86 @@
+// state.js
+
+var _ = require('underscore');
+
+var Lm = require('./Lm');
+
+var inputStates = {}; 
+var outputStates = {}; 
+
+// Define Vni class and its subclasses - InterNodeVni & IipVni
+function State() {
+  this.lm = Lm();
+}
+
+function NodePortVni( vnid ) {
+  State.call(this);
+}
+
+function IipState() {
+  State.call(this);
+}
+
+var IIP_PORT =  '*';
+ 
+module.exports = {
+
+    /**
+     * return a hash of all input states for every port and its sources 
+     */
+    allInputStates: function() { 
+        return inputStates;
+    },
+
+    /**
+     * return a hash of all input states for every port and its sources 
+     */
+    allOutputStates: function() { 
+        return outputStates;
+    },
+
+    /**
+     * Get/set the current state for the specified input port from the originating port
+     */
+    inputState: function( inputPortName, originatingNodeId, originatingPortName, payload ) {
```

Comment:
To avoid confusion with NoFlo payloads, I suggest renaming payload -> newState, where newState is a State object (with lm and data properties).

Also, how about renaming originatingNodeId -> originatingNodeName (for consistency with inputPortName and originatingPortName)?